src/deadline/client/cli/_groups/asset_group.py

```diff
@@ -155,3 +260,146 @@ def asset_download(**args):
+            continue
+        input_paths.append(Path(root_path, base_manifest_path.path))
+
+    return get_file_changes(
+        asset_manager=asset_manager, input_paths=input_paths, root_path=root_path, update=update
+    )
+
+
+def get_file_changes(
+    asset_manager: S3AssetManager, input_paths: List[Path], root_path: str, update: bool
+) -> List[(tuple)]:
+    """
+    Checks a manifest file, compares it to specified root directory or manifest of files with the local hash cache.
+    Returns a list of tuples containing the file information, and its corresponding file status.
+    """
+    cache_config = config_file.get_cache_directory()
+
+    with HashCache(cache_config) as hash_cache:
+        with concurrent.futures.ThreadPoolExecutor() as executor:
+            futures = {
+                executor.submit(
+                    asset_manager._process_input_path,
+                    path=path,
+                    root_path=root_path,
+                    hash_cache=hash_cache,
+                    update=update,
+                ): path
+                for path in input_paths
+            }
+            new_or_modified_paths: List[(tuple)] = []
+            for future in concurrent.futures.as_completed(futures):
+                (file_status, _, manifestPath) = future.result()
+                if file_status is FileStatus.NEW or file_status is FileStatus.MODIFIED:
+                    new_or_modified_paths.append((file_status, manifestPath))
+
+            return new_or_modified_paths
+
+
+def update_manifest(manifest: str, new_or_modified_paths: List[(tuple)]) -> BaseAssetManifest:
+    """
+    Updates the local manifest file to reflect modified or new files
+    """
+    for filename in os.listdir(manifest):
+        if filename.endswith("_input"):
+            manifest_file_path = os.path.join(manifest, filename)
+            with open(manifest_file_path, "r") as manifest_file:
+                manifest_data_str = manifest_file.read()
+                local_base_asset_manifest = decode_manifest(manifest_data_str)
+
+    # maps paths of local to optimize updating of manifest entries
+    manifest_info_dict = {
+        base_manifest_path.path: base_manifest_path
+        for base_manifest_path in local_base_asset_manifest.paths
+    }
+
+    for _, base_asset_manifest in new_or_modified_paths:
+        if base_asset_manifest.path in manifest_info_dict:
+            # Update the hash_value of the existing object
+            manifest_info_dict[base_asset_manifest.path].hash = base_asset_manifest.hash
+        else:
+            # Add the new object if it doesn't exist
+            manifest_info_dict[base_asset_manifest.path] = base_asset_manifest
+
+    # write ot local manifest
+    updated_path_list = list(manifest_info_dict.values())
+    local_base_asset_manifest.paths = updated_path_list
+    with open(manifest_file_path, "w") as manifest_file:
+        manifest_file.write(local_base_asset_manifest.encode())
```

Comment:
This looks incorrect. The `local_base_asset_manifest` file is being updated for each file from `os.listdir(manifest)` but we break out of that for loop and only operate on the last value read into the variable. Should the code underneath (starting at line 386) be indented another level?

Reply:
spoken offline, will include checks for 1 manifest